habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -386,6 +386,286 @@ def step(self, *args, is_last_action, **kwargs):
             return {}
 
 
+class Controller:
+    def __init__(self, v_max_default, w_max_default, track_yaw=True):
+        """PID controller, generate velocity command from the waypoints"""
+        self.track_yaw = track_yaw
+
+        # Params
+        self.v_max = v_max_default
+        self.w_max = w_max_default
+
+        self.lin_error_tol = self.v_max / 120
+        self.ang_error_tol = self.w_max / 120
+
+        self.acc_lin = 2.4
+        self.acc_ang = 2.4
+
+        # Init
+        self.xyt_goal = np.zeros(3)
```

Comment:
Does xyt correspond to X, Y and Theta? I would prefer the variable name to be `base_pose_goal`.